python/cuml/test/test_fil.py

```diff
@@ -560,7 +576,7 @@ def test_lightgbm(tmp_path, num_classes, n_categorical):
         # binary classification
         gbm_proba = bst.predict(X)
         fil_proba = fm.predict_proba(X)[:, 1]
-        gbm_preds = (gbm_proba > 0.5)
+        gbm_preds = (gbm_proba > 0.5).astype(int)
```

Comment:
As discussed offline, it's better to convert everything to float, and don't have to keep track of too many types.